Expensify/App
# Add IOU badge to report row and header

### Details
Adds IOU badge.

**Notes:**
- We still aren't yet handling the Pusher event so once the IOU is paid things do not update in realtime. This can probably be fixed in a later PR.
- The report header text wraps weirdly now that there is an extra element. I forced it to use ellipsis instead since I think it looks better.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/158424

### Tests
1. Create a new IOU transaction via
```
CreateIOUTransaction
authToken: <auth token>
debtorEmail: <email of user you want to request payment from>
currency: USD
comment: 'test'
amount: 299
```
2. Open E.cash as the user (same authToken that you just used)
3. Verify that there is now a badge to in the report row and header
4. Run the next command to mark it as paid
```
PayIOU
authToken: <auth token of the payee> // must sign in as the "debtor"
reportID: <IOU Report ID> 
paymentMethodType: 'Venmo'
```
5. Refresh the page
6. Verify that the badges are no longer displayed

### Tested On


### Screenshots


#### Web
<img width="1440" alt="Screen Shot 2021-03-26 at 7 57 47 AM" src="https://user-images.githubusercontent.com/32969087/112674069-91638380-8e09-11eb-9fcf-3570b5912d9d.png">

#### Mobile Web
<img width="371" alt="Screen Shot 2021-03-26 at 8 09 37 AM" src="https://user-images.githubusercontent.com/32969087/112674984-af7db380-8e0a-11eb-9105-6ec2819aa75e.png">
<img width="370" alt="Screen Shot 2021-03-26 at 8 09 17 AM" src="https://user-images.githubusercontent.com/32969087/112674992-b1477700-8e0a-11eb-97c3-b17e6eb0a226.png">

#### Desktop
<img width="1200" alt="Screen Shot 2021-03-26 at 8 11 58 AM" src="https://user-images.githubusercontent.com/32969087/112675224-f9669980-8e0a-11eb-8539-735ad4692aec.png">

#### iOS
<img width="396" alt="Screen Shot 2021-03-26 at 8 01 16 AM" src="https://user-images.githubusercontent.com/32969087/112674030-86a8ee80-8e09-11eb-8f19-1b2ec453b08e.png">
<img width="398" alt="Screen Shot 2021-03-26 at 8 01 06 AM" src="https://user-images.githubusercontent.com/32969087/112674043-8872b200-8e09-11eb-9f56-ff68341550cf.png">

#### Android
<img width="409" alt="Screen Shot 2021-03-26 at 8 07 59 AM" src="https://user-images.githubusercontent.com/32969087/112675013-b73d5800-8e0a-11eb-9e11-d47c71aed367.png">
<img width="399" alt="Screen Shot 2021-03-26 at 8 07 50 AM" src="https://user-images.githubusercontent.com/32969087/112675025-b99fb200-8e0a-11eb-8129-99f6b5fd0fa8.png">
